nucypher/blockchain/eth/agents.py

```diff
@@ -806,7 +806,7 @@ def fetch_arrangement_addresses_from_policy_txid(self, txhash: Union[str, bytes]
         # TODO: Won't it be great when this is impossible?  #1274
         _receipt = self.blockchain.client.wait_for_receipt(txhash, timeout=timeout)
         transaction = self.blockchain.client.w3.eth.getTransaction(txhash)
-        _signature, parameters = self.contract.decode_function_input(transaction.data)
+        _signature, parameters = self.contract.decode_function_input(transaction.input)
```

Comment:
Im trying to isolate the upstream API version change to reference.

Reply:
Ok here is a small lead: <https://web3py.readthedocs.io/en/latest/web3.eth.html#web3.eth.Eth.getTransaction>

This call executes an RPC call `eth_getTransaction`, which may need to be composed into the client layer for cross-compatibility.

Reply:
hmm well, these docs show no "data" key, which... seems consistent.

Reply:
As we've seen before, different ethereum client can have slightly different return values.  As a matter of design, all of our RPC calls are routed through clients, but this one bypasses that compatibility layer resulting in the CI failure.

Reply:
I'd say that this is something that the Web3 layer should handle. Not that we shouldn't do nothing, but I think it's an inconsistency on their side. If we can confirm this is the case, we should open an issue there.

Reply:
I'd argue that we already accommodate eth-tester in this regard, perhaps the issue is directed there?

Reply:
once I get the tests passing, I was planning on building this into our Clients which should be less janky than what I'm doing.

Reply:
:+1: - Seems like those two are actually the same effort!